main.go

```diff
@@ -169,18 +170,42 @@ func binaries(filter map[string]matched) ([]string, error) {
 			if strings.HasPrefix(fi.Name(), ".") {
 				continue
 			}
+			// Ignore non-exe files on Windows.
+			if fi.Name() != binaryName(canonicalName(fi.Name())) {
+				continue
+			}
 
 			// If user specified a list of binaries, filter out binaries that don't match.
 			if len(filter) != 0 {
-				if _, ok := filter[fi.Name()]; !ok {
+				if _, ok := filter[canonicalName(fi.Name())]; !ok {
 					continue
 				}
-				filter[fi.Name()] = matched(true)
+				filter[canonicalName(fi.Name())] = matched(true)
 			}
 
-			binaries = append(binaries, fi.Name())
+			binaries = append(binaries, canonicalName(fi.Name()))
 		}
 	}
 
 	return binaries, nil
 }
+
+// canonicalName, when called on a Windows system, trims the ".exe" suffix from
+// the end of a binary's filename.
+func canonicalName(anyName string) string {
+	if "windows" == runtime.GOOS {
+		return strings.TrimSuffix(anyName, ".exe")
+	} else {
+		return anyName
+	}
+}
```

Comment:
Instead of `canonicalName`, create a func:

```Go
// commandName returns the name of Go command that would've resulted in this binary, if possible.
func commandName(binaryName string) (commandName string, ok bool) {
	if "windows" == runtime.GOOS {
		if !strings.HasSuffix(binaryName, ".exe") {
			return "", false
		}
		return binaryName[:len(binaryName)-4], true
	}
	return binaryName, true
}
```

Now that I think about it, we might as well move the two other checks (`IsDir` and if it begins with a dot) into this func, by expanding its signature to accept `os.FileInfo`.